blog/designing-for-journey-candance-van-oostrum.md

```diff
@@ -0,0 +1,35 @@
+
+As companies grow, the service topology, dependencies and complexities grow hand-in-hand. Along with this growth, comes increased debt (both ux and tech), high on-call volumes, constant interruptions, unplanned work, and weak or unreliable institutional knowledge. Not only do these issues surface in the UX, but it’s often put on the product design teams to fix them, without fixing the underlying problems.
+
+For example, I was once working with a prod-eng team to improve the usability of a core feature and we found, to our surprise, that when the user finished a 3 step create flow, the object wasn’t immediately created. The user was redirected back to the landing page without receiving a success message or any reference of the newly created item, because it in fact had not yet been created due to our adoption of eventual consistency. Unbeknown to the user, the request went into a queue and would eventually be created, and the user would proceed to re-try to create the item. .
+
+What happens next depends on how long it takes for the queue to process the request. In most cases, by the time the user tried to re-enter the name input, the inline validation would throw an error that the name already exists. We observed that users would slightly change the name to satisfy the error message and continue, now adding a 2nd item into the queue and re-entering this loop. In other scenarios, users would move to the final step in the process, hit the create button and would get some gnarly errors because the name already exists and wasn’t caught on the first step, because it was still in the queue.
+
+We pulled 90 days of historical data to understand what was actually causing delays for processing the request. We found that the majority of requests took between 3-8 seconds but could potentially last up to 5 minutes, and in rare circumstances even longer. With this data in hand, we started to ideate on different ways to handle this delay in the UI. We could definitely hold the button with a spinner to solve the 3 second delay, but that wouldn’t work for a 5 minute request. Anything longer than a few seconds and the user would assume the app had frozen. Due to the fact that we didn’t know how long the delay would be, we couldn’t actually design for different time ranges.
```

Comment:
```suggestion
We pulled 90 days of historical data to understand what was actually causing delays for processing the request. We found that the majority of requests took 3 to 8 seconds but could potentially last up to 5 minutes, and in rare circumstances even longer. With this data in hand, we started to ideate on different ways to handle this delay in the UI. We could definitely hold the button with a spinner to solve the 3-second delay, but that wouldn’t work for a 5-minute request. Anything longer than a few seconds and the user would assume the app had frozen. Because we didn’t know how long the delay would be, we couldn’t actually design for various time ranges.
```